### PR TITLE
Renamed NextInputs and GetStart by advance_iterators - issue 152

### PR DESCRIPTION
**Pull request by AndresSanchezCuadrado**
Renamed NextInputs and GetStart function names by advance_iterators.

### CODE REVIEW

File: include/native/reduce.h [hunk 1]
```diff
@@ -239,7 +239,7 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
  void Reduce( InputIt first, InputIt last, OutputIt firstOut, Operation && op, MoreIn ... inputs ) {
     while( first != last ) {
         *firstOut = op( *first, *inputs ... );
-        NextInputs( inputs... );
+        advance_iterators( inputs... );
```
Comment (jdgarciauc3m) [inline include/native/reduce.h]: Is this fixing a commented code?

File: include/native/mapreduce.h [hunk 2]
```diff
@@ -83,7 +83,7 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
  void Reduce( InputIt first, InputIt last, OutputIt firstOut, Operation && op, MoreIn ... inputs ) {
     while( first != last ) {
         *firstOut = op( *first, *inputs ... );
-        NextInputs( inputs... );
+        advance_iterators( inputs... );
```
Comment (jdgarciauc3m) [inline include/native/mapreduce.h]: Is this fixing commented code?

File: include/native/stencil.h [hunk 3]
```diff
@@ -93,12 +93,12 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
                // Register the thread in the execution model
                p.register_thread();
                
-               GetStart(n, i, inputs ...);
+               advance_iterators(n, i, inputs ...);
```
Comment (jdgarciauc3m) [inline include/native/stencil.h]: This needs to be reviewed by  @Drioastorga

File: include/omp/divideandconquer.h [hunk 4]
```diff
@@ -138,7 +138,7 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
  void Reduce( InputIt first, InputIt last, OutputIt firstOut, Operation && op, MoreIn ... inputs ) {
     while( first != last ) {
         *firstOut = op( *first, *inputs ... );
-        NextInputs( inputs... );
+        advance_iterators( inputs... );
```
Comment (jdgarciauc3m) [inline include/omp/divideandconquer.h]: Fixing commented code?

File: include/omp/reduce.h [hunk 5]
```diff
@@ -138,7 +138,7 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
  void Reduce( InputIt first, InputIt last, OutputIt firstOut, Operation && op, MoreIn ... inputs ) {
     while( first != last ) {
         *firstOut = op( *first, *inputs ... );
-        NextInputs( inputs... );
+        advance_iterators( inputs... );
```
Comment (jdgarciauc3m) [inline include/omp/reduce.h]: Fixing commented code?

File: include/omp/stencil.h [hunk 6]
```diff
@@ -86,12 +86,12 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
 
            auto out = firstOut + (elemperthr * i);
         
-           GetStart(elemperthr, i, inputs ...);
+           advance_iterators(elemperthr, i, inputs ...);
```
Comment (jdgarciauc3m) [inline include/omp/stencil.h]: Needs to be reviewed by @Drioastorga

File: include/seq/divideandconquer.h [hunk 7]
```diff
@@ -52,7 +52,7 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
  void Reduce( InputIt first, InputIt last, OutputIt firstOut, Operation && op, MoreIn ... inputs ) {
     while( first != last ) {
         *firstOut = op( *first, *inputs ... );
-        NextInputs( inputs... );
+        advance_iterators( inputs... );
```
Comment (jdgarciauc3m) [inline include/seq/divideandconquer.h]: Fixing commented code?

File: include/seq/mapreduce.h [hunk 8]
```diff
@@ -71,7 +71,7 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
  void Reduce( InputIt first, InputIt last, OutputIt firstOut, Operation && op, MoreIn ... inputs ) {
     while( first != last ) {
         *firstOut = op( *first, *inputs ... );
-        NextInputs( inputs... );
+        advance_iterators( inputs... );
```
Comment (jdgarciauc3m) [inline include/seq/mapreduce.h]: Fixing commented code?

File: include/seq/reduce.h [hunk 9]
```diff
@@ -81,7 +81,7 @@ template <typename InputIt, typename OutputIt, typename ... MoreIn, typename Ope
  void Reduce( InputIt first, InputIt last, OutputIt firstOut, Operation && op, MoreIn ... inputs ) {
     while( first != last ) {
         *firstOut = op( *first, *inputs ... );
-        NextInputs( inputs... );
+        advance_iterators( inputs... );
```
Comment (jdgarciauc3m) [inline include/seq/reduce.h]: Fixing commented code?

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
In many cases we are updating commented out code, that needs to be removed anyway.

In general we need that @Drioastorga reviews the change-set.

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
I still see a lot of modifications to commented code.

This sould not be part of this pull-request.

**Comment by jdgarciauc3m** [comment]
Added @Drioastorga as additional reviewer of this PR. We will wait his approval for merge.

**Comment by jdgarciauc3m** [comment]
Still this needs review by @Drioastorga

**Review by Drioastorga — APPROVED** [review]
It works perfectly and the changes are right.

**Review by jdgarciauc3m — APPROVED** [review]

**Comment by jdgarciauc3m** [comment]
Solves issue #152

**Pull request merged**